base/dsfs/dataset.go

```diff
@@ -241,17 +248,30 @@ var Timestamp = func() time.Time {
 	return time.Now().UTC()
 }
 
+// BodyAction represents the action that should be taken to understand how the body changed
+type BodyAction string
+
+const (
+	// BodyDefault is the default action: compare them to get how much changed
+	BodyDefault = BodyAction("default")
+	// BodySame means that the bodies are the same, no need to compare
+	BodySame = BodyAction("same")
+	// BodyTooBig means the body is too big to compare, and should be assumed to have changed
+	BodyTooBig = BodyAction("too_big")
```

Comment:
fan of this. Helps me think about FSI & transform script interactions where a file exists on disk that we don't want to use if unchanged. Don't think we can solve that here, but enumerating what actions we can take is a nice technique